src/Microsoft.TestPlatform.Build/Tasks/TestTaskUtils.cs

```diff
@@ -0,0 +1,222 @@
+            isRunSettingsEnabled = true;
+            builder.AppendSwitchIfNotNull("--settings:", task.VSTestSetting);
+        }
+
+        if (task.VSTestTestAdapterPath != null && task.VSTestTestAdapterPath.Any())
+        {
+            foreach (var arg in task.VSTestTestAdapterPath)
+            {
+                builder.AppendSwitchIfNotNull("--testAdapterPath:", arg);
+            }
+        }
+
+        builder.AppendSwitchIfNotNull("--framework:", task.VSTestFramework);
+
+        // vstest.console only support x86 and x64 for argument platform
+        if (!task.VSTestPlatform.IsNullOrEmpty() && !task.VSTestPlatform.Contains("AnyCPU"))
+        {
+            builder.AppendSwitchIfNotNull("--platform:", task.VSTestPlatform);
+        }
+
+        builder.AppendSwitchIfNotNull("--testCaseFilter:", task.VSTestTestCaseFilter);
+
+        if (task.VSTestLogger != null && task.VSTestLogger.Any())
+        {
+            foreach (var arg in task.VSTestLogger)
+            {
+                builder.AppendSwitchIfNotNull("--logger:", arg);
+
+                if ((arg != null) && arg.StartsWith("console", StringComparison.OrdinalIgnoreCase))
+                {
+                    isConsoleLoggerSpecifiedByUser = true;
+                }
+            }
+        }
+
+        builder.AppendSwitchIfNotNull("--resultsDirectory:", task.VSTestResultsDirectory);
+
+        if (task.VSTestListTests)
+        {
+            builder.AppendSwitch("--listTests");
+        }
+
+        builder.AppendSwitchIfNotNull("--diag:", task.VSTestDiag);
+
+        if (task.TestFileFullPath == null || task.TestFileFullPath.ItemSpec.IsNullOrEmpty())
+        {
+            task.Log.LogError(Resources.Resources.TestFilePathCannotBeEmptyOrNull);
+        }
+        else
+        {
+            builder.AppendFileNameIfNotNull(task.TestFileFullPath);
+        }
+
+        // Console logger was not specified by user, but verbosity was, hence add default console logger with verbosity as specified
+        if (!task.VSTestVerbosity.IsNullOrWhiteSpace() && !isConsoleLoggerSpecifiedByUser)
+        {
+            var normalTestLogging = new List<string>() { "n", "normal", "d", "detailed", "diag", "diagnostic" };
+            var quietTestLogging = new List<string>() { "q", "quiet" };
+
+            string vsTestVerbosity = "minimal";
+            if (normalTestLogging.Contains(task.VSTestVerbosity.ToLowerInvariant()))
+            {
+                vsTestVerbosity = "normal";
+            }
+            else if (quietTestLogging.Contains(task.VSTestVerbosity.ToLowerInvariant()))
+            {
+                vsTestVerbosity = "quiet";
+            }
+
+            builder.AppendSwitchUnquotedIfNotNull("--logger:", $"Console;Verbosity={vsTestVerbosity}");
+        }
+
+        if (task.VSTestBlame || task.VSTestBlameCrash || task.VSTestBlameHang)
+        {
+            var dumpArgs = new List<string>();
+            if (task.VSTestBlameCrash || task.VSTestBlameHang)
```

Comment:
This outer if seems un-needed as it only contains 2 ifs checking each side of this outer if condition.

Reply:
It is useful when `VSTestBlame` is `true` and the others are `false`: it adds the `--Blame` switch.